buildkitd/buildkitd.go

```diff
@@ -172,13 +173,21 @@ func ResetCache(ctx context.Context, console conslogging.ConsoleLogger, image, c
 // that can be used to connect to it.
 func maybeStart(ctx context.Context, console conslogging.ConsoleLogger, image, containerName, installationName string, fe containerutil.ContainerFrontend, settings Settings, opts ...client.ClientOpt) (cinfo *client.Info, winfo *client.WorkerInfo, finalErr error) {
 	if settings.StartUpLockPath != "" {
+		var tryLockDone atomic.Bool
+		go func() {
+			time.Sleep(3 * time.Second)
```

Comment:
I think this will leak a goroutine (for 3 seconds). We could alternately select on some done channel that we close somewhere, but :shrug:.